src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableEntityContext.cs

```diff
@@ -101,7 +169,19 @@ TState IDurableEntityContext.GetState<TState>(Func<TState> initializer)
 
             if (!this.StateWasAccessed)
             {
-                TState defaultValue = initializer != null ? initializer() : default(TState);
+                TState defaultValue = default(TState);
+
+                if (initializer != null)
+                {
+                    try
+                    {
+                        defaultValue = initializer();
+                    }
+                    catch (Exception e)
+                    {
+                        throw new EntitySchedulerException($"failed to construct entity state: {e.Message}", e);
```

Comment:
nit: "failed" should be capitalized. You may also want to prefer "initialize" over "construct" to match the name of the function that threw the exception.

Reply:
ok, I will to that for the other 5 occurrences as well.